src/InputFileResolver.ts

```diff
@@ -36,12 +37,12 @@ export default class InputFileResolver {
     console.log(`WARNING: Globbing expression "${expression}" did not result in any files.`)
   }
 
-  private static resolveFileGlobs(sourceExpressions: string[], resultFiles: string[]): Promise<void[]> {
-    return Promise.all(sourceExpressions.map((mutateFileExpression: string) => glob(mutateFileExpression).then(files => {
+  private static resolveFileGlobs(sourceExpressions: string[]): Promise<string[]> {
+    return Promise.all(sourceExpressions.map(expression => glob(expression).then(files => {
       if (files.length === 0) {
-        this.reportEmptyGlobbingExpression(mutateFileExpression);
+        this.reportEmptyGlobbingExpression(expression);
       }
-      files.forEach(f => resultFiles.push(f));
-    })));
+      return files;
+    }))).then(files => _.flatten(files));
```

Comment:
This is the new way of returning an array. Per `expression` a globbing promise get's resolved. The promises get created in the correct order, but the resolving can still happen in any order. However the `Promise.all` api will handle this for us and give us a `files: string[][]` in the correct order. All we have to do is flatten it before returning.

It seems to work for a test project i used.